### PR TITLE
Add upserting and deletion capability to PGVector

### PR DESCRIPTION
**Pull request by angelol**
Currently, the PGVector vectorstore only supports adding embeddings to the store. This PR adds upserting and deletion capability to the store, so you can update existing embeddings if the underlying text have changed. This is useful if you have indexed a set of documents but some of the documents are being changed and you want to keep the index up-to-date.

To use this feature, simply provided ids for all texts/or documents. The pgvector notebook has been updated accordingly.

The fixes issue #6866. 

👀  @rlancemartin, @eyurtsev

In case you want to announce on Twitter, my Twitter handle is @angelol

### CODE REVIEW

**Comment by leo-gan** [comment]
@angelol Hi , could you, please, resolve the merging issues? After that ping me and I push this PR for the review. Thanks!

File: langchain/vectorstores/pgvector.py [hunk 1]
```diff
@@ -213,25 +223,36 @@ def add_embeddings(
...
         if not metadatas:
             metadatas = [{} for _ in texts]
 
+        last_seen_id = None
         with Session(self._conn) as session:
             collection = self.get_collection(session)
             if not collection:
                 raise ValueError("Collection not found")
             for text, metadata, embedding, id in zip(texts, metadatas, embeddings, ids):
+                if should_upsert and id != last_seen_id:
```
Comment (eyurtsev) [inline langchain/vectorstores/pgvector.py:248]: This part is quite inefficient -- a network request per upsert

File: langchain/vectorstores/pgvector.py [hunk 2]
```diff
@@ -205,6 +206,15 @@ def __from(
 
         return store
 
+    def delete_embeddings(self, ids: List[str]) -> None:
```
Comment (eyurtsev) [inline langchain/vectorstores/pgvector.py:209]: Not sure that deletion of embeddings should be supported. Likely a lot of things expect the embeddings to exist. An update in place / upsert should be sufficient instead

**Review by eyurtsev — CHANGES_REQUESTED** [review]

**Comment by efriis** [comment]
Closing because the PR wouldn't line up with the current directory structure of the library (would need to be in /libs/langchain/langchain instead of /langchain). Feel free to reopen against the current head if it's still relevant!

**Pull request closed without merging**